utils/remote/common.py

```diff
@@ -117,7 +123,11 @@ def _connect_transport(self, transport, username=None, password=None):
             hostkeys.save(hostkey_file)
 
         try:
-            transport.auth_password(username, password)
+            try:
+                key = RSAKey.from_private_key(password)
+                transport.auth_publickey(username, key)
+            except AttributeError:
+                transport.auth_password(username, password)
```

Comment:
I'm not happy about it either. I think I'm going to add a `Credential` class instead for both username and password. I don't want to introduce yet another parameter, it would need to be passed through in a lot of different places, and there could very well be different method of authentication later (like talking to a SSH agent).

As for passing a file object - the key isn't necessarily a file on disk. Maybe it would be easier to pass the contents as a byte string?

Reply:
> As for passing a file object - the key isn't necessarily a file on disk.

Sorry I don't follow. We get the file object from the pathname through `open`. Why can't we use the pathname instead of the file object and use `from_private_key_file` instead of `from_private_key` here?

Reply:
A file object is not necessarily backed by an actual file on disk - it might as well be a StringIO, a pipe or whatever, and we should support these use cases - so either leave it a file, or make it a byte string and make the caller read the file.

Reply:
Not convinced but I'm ok with it.

Reply:
Assuming we want to store private keys in a database in the future (for redundancy, because local state is bad). If we passed a filename, we would have to write out the key to a temporary file.

How about passing a byte string? That would leave it entirely up to the caller.

Reply:
> How about passing a byte string? That would leave it entirely up to the caller.

Sounds good!